### PR TITLE
GODRIVER-2883 Handle boolean 'ok' responses

### PR DESCRIPTION
**Pull request by blink1073**
GODRIVER-2883

## Summary
Handle boolean "ok" messages in command responses, and add text.

## Background & Motivation
While the specification says that [Command Responses](https://www.mongodb.com/docs/manual/reference/method/db.runCommand/#std-label-command-response) have an "ok" field that "A number that indicates whether the command has succeeded (1) or failed (0).", there are some cases where the returned command can have a boolean value.   Update the Go driver to handle boolean "ok" responses and treat "ok": true as a success. 

We do not handle the case of a "hello" command returning a boolean value, since it has not been observed and we do not have a direct way to test the handling of hello responses without creating a new unified spec test.

### CODE REVIEW

File: x/mongo/driver/auth/plain_test.go [hunk 1]
```diff
@@ -145,3 +145,48 @@ func TestPlainAuthenticator_Succeeds(t *testing.T) {
...
+	c := &drivertest.ChannelConn{
+		Written:  make(chan []byte, 1),
+		ReadResp: resps,
+		Desc:     desc,
+	}
+
+	err := authenticator.Auth(context.Background(), &Config{Description: desc, Connection: c})
+	if err != nil {
+		t.Fatalf("expected no error but got \"%s\"", err)
+	}
```
Comment (matthewdale) [inline x/mongo/driver/auth/plain_test.go:179]: Optional: Consider using `require.NoError` instead.

E.g.
```go
require.NoError(t, err, "Auth error")
```

File: x/mongo/driver/auth/plain_test.go [hunk 2]
```diff
@@ -145,3 +145,48 @@ func TestPlainAuthenticator_Succeeds(t *testing.T) {
...
+
+	if len(c.Written) != 1 {
+		t.Fatalf("expected 1 messages to be sent but had %d", len(c.Written))
+	}
```
Comment (matthewdale) [inline x/mongo/driver/auth/plain_test.go:183]: Optional: Consider using `require.Len` instead.

E.g.
```go
require.Len(t, c.Written, 1, "expected 1 messages to be sent")
```

File: x/mongo/driver/auth/plain_test.go [hunk 3]
```diff
@@ -145,3 +145,48 @@ func TestPlainAuthenticator_Succeeds(t *testing.T) {
...
+
+	payload, _ := base64.StdEncoding.DecodeString("AHVzZXIAcGVuY2ls")
```
Comment (matthewdale) [inline x/mongo/driver/auth/plain_test.go:185]: Optional: Not handling this error may cause confusion if anyone ever updates the test and provides an invalid base64 string. Consider handling the error to make failures more obvious for future maintainers.

E.g.
```go
payload, err := base64.StdEncoding.DecodeString("AHVzZXIAcGVuY2ls")
require.NoError(t, err, "DecodeString error")
```

**Review by matthewdale — APPROVED** [review]
Looks good 👍

File: x/mongo/driver/auth/plain_test.go (hunk 1 above)
Comment (blink1073) [reply x/mongo/driver/auth/plain_test.go:179]: Done

File: x/mongo/driver/auth/plain_test.go (hunk 2 above)
Comment (blink1073) [reply x/mongo/driver/auth/plain_test.go:183]: Done

File: x/mongo/driver/auth/plain_test.go (hunk 3 above)
Comment (blink1073) [reply x/mongo/driver/auth/plain_test.go:185]: Done

**Review by qingyang-hu — APPROVED** [review]

**Pull request merged**